# STAT 207 Lab 7: Hypothesis Testing - [25.5 total points] (0.5) is bonus points - Solutions

Due: Wednesday, March 24 11:59pm CST



## <u>Case Study</u>: U.S Unemployment Rates and Median Education Levels
In this lab we will explore the statistical concepts that we've learned this week while also examining the **unemployment rates** and **median education levels** of the counties in the U.S. 

### Imports

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns; sns.set()

## 1. <u>Preliminaries</u>

### Read the county.csv into a dataframe, call it df, and display the first five rows.

Remember that 'data unavailable' represents missing values in this csv. Make sure you encode these values as NaN when you read the data in.

In [4]:
df = pd.read_csv("county.csv")
df = df.replace('data unavailable','NaN')
df.head()

name    state pop2000  pop2010 pop2017 pop_change poverty  \
0  Autauga County  Alabama   43671    54571   55504       1.48    13.7   
1  Baldwin County  Alabama  140415   182265  212628       9.19    11.8   
2  Barbour County  Alabama   29038    27457   25270      -6.22    27.2   
3     Bibb County  Alabama   20826    22915   22668       0.73    15.2   
4   Blount County  Alabama   51024    57322   58013       0.68    15.6   

   homeownership  multi_unit unemployment_rate metro    median_edu  \
0           77.5         7.2              3.86   yes  some_college   
1           76.7        22.6              3.99   yes  some_college   
2           68.0        11.1               5.9    no    hs_diploma   
3           82.9         6.6              4.39   yes    hs_diploma   
4           82.0         3.7              4.02   yes    hs_diploma   

  per_capita_income median_hh_income smoking_ban  
0           27841.7            55317        none  
1          27779.85            52562        none  
2          17891.73            33368     partial  
3          20572.05            43404        none  
4          21367.39            47412        none

### Create a series that is comprised of just the unemployment rate variable and drop any missing values from this dataset.

In [5]:
missing_values = ['data unavailable']
df_pew = pd.read_csv("county.csv", na_values=missing_values)
df_unemploy = df_pew["unemployment_rate"].dropna()
df_unemploy

0       3.86
1       3.99
2       5.90
3       4.39
4       4.02
        ... 
3137    4.55
3138    2.99
3139    4.50
3140    4.08
3141    3.98
Name: unemployment_rate, Length: 3139, dtype: float64

### Create a series that is comprised of just the median education variable and drop any missing values from this dataset.

In [20]:
df_median_edu = df_pew["median_edu"].dropna()
df_median_edu

0       some_college
1       some_college
2         hs_diploma
3         hs_diploma
4         hs_diploma
            ...     
3137    some_college
3138       bachelors
3139    some_college
3140    some_college
3141    some_college
Name: median_edu, Length: 3140, dtype: object

## 2. <u>Unemployment Rate Population Analysis</u>

We will assume that the series of all county unemployment rates from 1b is our **population**. Usually, we do not have the whole population at our disposal, which is why we conduct inference in the first place.

However, when conducting our hypothesis testing in this assignment we will assume that we don't have this population information. After we have conducted the hypothesis testing, we can use our population information to "check" our answers to learn more about how hypothesis testing works.

### 2a. [0.5 pt] Calculate the mean and standard deviation unemployment rate of ALL U.S. counties.

In [22]:
df_unemploy.mean()

4.611443134756292

In [23]:
df_unemploy.std()

1.648761085622538

### 2b.  [1 pt] Calculate the z-score of the unemployment rate of 'Champaign County' in the state of 'Illinois'.

In [34]:
df_pew[(df['name'] == 'Champaign County') & (df['state'] == 'Illinois')]['unemployment_rate']

604    4.24
Name: unemployment_rate, dtype: float64

In [47]:
z_score = (4.24 - df_unemploy.mean()) / df_unemploy.std()
z_score

-0.2252862091392959

## 3. <u>Unemployment Rate Sample Analysis</u>

### 3a.  [0.5 pt] Collect a random sample of size 35 from your population of all county unemployment rates (without replacement) using a random state of 3.

In [8]:
df_sample=df_unemploy.sample(35,replace=False,random_state=3)


### 3b.  [0.5 pt] Calculate the mean and standard deviation of this sample.

In [9]:
df_sample.mean()

4.252

In [52]:
df_sample.std()

1.3095751443974222

## 4. <u>Unemployment Rate Sampling Distribution Analysis</u>

Assume you know $\sigma$ in this part.

### 4a.  [0.5 pt] What is the mean and standard deviation of the sampling distribution of sample mean unemployment rates (collected from samples of size 35)?

4.611

In [85]:
sample_dis_mean = df_unemploy.mean()
sample_dis_mean

4.611443134756292

0.2787

In [86]:
sample_dis_std = df_unemploy.std()/np.sqrt(35)
sample_dis_std

0.2786914893089392

### 4b.  [1 pt] Is the sampling distribution of samples means (collected from samples of size 35) approximately normal? Explain why or why not?

This sampling distribution of sample means will be approximately normal. This is because of the general rule of thumb indicating that samples of size 30 or greater will have a fairly normal distribution regardless of the shape of the distribution. The size of samples(35) is more than 30.

### 4c.  [1 pt] What is the z-score of our sample mean from 3b?

In [88]:
z_score2 = (df_sample.mean() - sample_dis_mean) / sample_dis_std
z_score2

-1.2897528218302958

### 4d.  [1 pt] What is the probability that a randomly selected sample mean z-score is greater than or equal to the z-score that we calculated in 4c?

In [58]:
from scipy.stats import norm
1 - norm.cdf(z_score2)

0.9014317534230412

### 4e.  [0.5 pt] What is the probability that a randomly selected sample mean z-score is less than or equal to the negative absolute value of the z-score that we calculated in 4c?

In [59]:
norm.cdf(-np.abs(z_score2))

0.0985682465769589

## 5. <u>Hypothesis Testing for the Average Unemployment Rate of ALL US Counties</u>

For now, suppose that we know nothing about our population that we explored in #2.


Suppose that a reliable source suggested that the average unemployment rate of all US counties is 4.611. We are skeptical of this source, and would like to test this claim and see if there is evidence to suggest that this is not the case. We will use our random sample that we collected from the population in #3 to test this claim.

### 5a.  [1 pt] Formulate your null and alternative hypothses for this hypothesis test.

null hypothses: mean equals 4.611
alternative hypothses: mean does not equal 4.611
where mean is he average unemployment rate of all US counties.

### 5b.  [1 pt] Check the conditions for conducting hypothesis testing using the sample from #3.



Condition 1:The observation is independent.
Because the sample is collected via random sampling and n < 10% of population of all US counties.
Condition 2: Either n>30 or the population distribution is normal
n = 35 > 30, so this condition is met.

### 5c.  [1 pt] Calculate the p-value for this test.

Assume you do not know $\sigma$.

In [60]:
assumed_mean = 4.611
std_error = df_unemploy.std()/np.sqrt(35)

In [61]:
p_value = 2 * (1 - norm.cdf(df_sample.mean(), loc=assumed_mean, scale = std_error))
p_value

1.8023106901828732

### 5d.  [1 pt] Make a conclusion about your hypotheses using this p-value and a significance level of $\alpha=0.05$.

Because pvalue = 1.80 >= a = 0.05, we fail to reject the null hypothesis. Thus, we don't have sufficient evidence to suggest the alternative hypothesis(the average unemployment rate of all US counties is different than 4.611).

### 5e.  [1 pt] Now we would like to make a 95% confidence interval for the average unemployment rate of al U.S. counties. First, check the conditions for creating this confidence interval.

The observations are normal (sample is collected via random sampling, n<10% of all U.S county homeownership rates)
The sample size > 30

### 5f.  [1 pt] Now create this 95% confidence interval.

Assume you do not know $\sigma$.

In [62]:
alpha = 0.05
critical_value = norm.ppf(1-alpha/2)
critical_value

1.959963984540054

In [64]:
lower_bound=df_sample.mean() - critical_value*(std_error)
upper_bound=df_sample.mean() + critical_value*(std_error)
print(lower_bound, "," ,upper_bound)

3.7057747181566496 , 4.79822528184335


### 5g.  [1 pt] Use this confidence interval to make a conclusion about your hypothesis tests.

Because the null value(4.611) is inside the confidence interval, we fail to reject the null hypothesis. Thus, we don't have sufficient evidence to suggest the alternative hypothesis(the average unemployment rate of all US counties is different than 4.611).

### 5h.  [0.5 pt] NOW, using what we *actually* know about the population and population mean, did our hypothesis test *actually* make the right conclusions in 5d and 5g?

The hypothesis is actually right, but the hypothesis tests cannot make the right conclusion.

## 6. <u>Median Education Level Population Analysis</u>

We will assume that the series of all county median education levels from 1c is our **population**. Usually, we do not have the whole population at our disposal, which is why we conduct inference in the first place.

However, when conducting our hypothesis testing in this assignment we will assume that we don't have this population information. After we have conducted the hypothesis testing, we can use our population information to "check" our answers to learn more about how hypothesis testing works.



### <u>Tutorial</u>: Conditional Filtering of a Series

In [4]:
import pandas as pd
import numpy as np
small_series=pd.Series(['stat107', 'stat207', 'stat307', 'stat207'])
small_series

0    stat107
1    stat207
2    stat307
3    stat207
dtype: object

We can count the number of entries in a series that meet a certain condition by setting up a condition upon the values in a series and taking the sum of the Boolean values.

In [5]:
small_series=='stat207'

0    False
1     True
2    False
3     True
dtype: bool

In [6]:
np.sum(small_series=='stat207')

2

### 6a.  [0.5 pt] Calculate the proportion of all U.S. counties that have a median education level of 'hs_diploma'.

In [67]:
np.sum(df_median_edu == 'hs_diploma') / df_median_edu.shape[0]

0.4449044585987261

## 7. <u>Median Population Sample Analysis</u>

### 7a.  [0.5 pt] Collect a random sample of 40 county median education levels from the population of ALL U.S. county median education levels, (without replacement), using a random state of 100.

In [71]:
df_sample2=df_median_edu.sample(40,replace=False,random_state=100)
df_sample2

1168      hs_diploma
1049    some_college
1921    some_college
1740    some_college
3138       bachelors
497     some_college
1745    some_college
1148    some_college
1248      hs_diploma
1333    some_college
1907    some_college
1738    some_college
117       hs_diploma
2711      hs_diploma
617     some_college
781       hs_diploma
1276    some_college
952       hs_diploma
2444      hs_diploma
2746    some_college
1669    some_college
1079      hs_diploma
2022    some_college
2342      hs_diploma
154       hs_diploma
3004      hs_diploma
2140      hs_diploma
29        hs_diploma
31        hs_diploma
322     some_college
2027    some_college
1638    some_college
357       hs_diploma
1989      hs_diploma
775     some_college
2089    some_college
3036      hs_diploma
2825      hs_diploma
1162      hs_diploma
1137      hs_diploma
Name: median_edu, dtype: object

### 7b.  [0.5 pt] Calculate the proportion of counties in this sample that have a median education level of high school diploma.

In [77]:
prop = np.sum(df_sample2 == 'hs_diploma') / 40
prop

0.5

## 8. <u>Sampling Distribution of Sample Proportions Analysis</u>



### 8a. [1 pt] What is the mean and standard deviation of the sampling distribution of sample propotion of counties with a median education level of high school diploma (collected from samples of size 40)?

Assume you actually know the population proprtion $p$ in part 8.

0.5

In [80]:
sample_dis_mean2 = prop

In [81]:
sample_dis_mean2

0.5

0.079

In [83]:
sample_dis_std2 = np.sqrt(prop*(1-prop)/40)
sample_dis_std2

0.07905694150420949

### 8b.  [1 pt] What is the z-score of our sample proportion from 7b?

In [87]:
z_score3 = (prop - sample_dis_mean2) / sample_dis_std
z_score3

0.0

## 9. <u>Hypothesis Testing for the Proportion of ALL U.S. Counties that have a Median Education Level of High School Diploma</u>

For now, suppose that we know nothing about our population that we explored in #6.

Previous studies have suggested that in the past, 45% of U.S. counties have a median education level of a high school diploma. We are skeptical that that is the still the case. We would like to test (using our sample from #7), whether there is evidence to suggest that the the proportion of counties with a median education level of high school diploma is diffrent from 45%.

### 9.a.  [1 pt] Formulate your null and alternative hypothses for this hypothesis test.

null hypothses: p equals 45%
alternative hypothses: p does not equal 45%
where p is the proportion of U.S. counties have a median education level of a high school diploma

### 9b.  [1 pt] Check the conditions for conducting hypothesis testing using the sample from #7.



Condition 1:The observation is independent.
Because the sample is collected via random sampling and n < 10% of population of all US counties.
Condition 2: np >= 10 and n(1-p) >= 10
np = 40 * 45% >= 10, n(1-p) = 40 * (1 - 45%) >= 10.

### 9c.  [1 pt] Calculate the p-value for this test.

Assume you do not know anything about the population proportion (including the value of $p$).

In [89]:
assumed_prop = 0.45

In [90]:
std_error2 = np.sqrt(assumed_prop*(1-assumed_prop)/40)

In [91]:
p_value2 = 2*(norm.cdf(0.5, loc=assumed_prop, scale=std_error2))

In [92]:
p_value2

1.4749900335813966

1.47

### 9d.  [1 pt] Make a conclusion about your hypotheses using this p-value and a significance level of $\alpha=0.10$.

Because pvalue = 1.47 >= a = 0.10, we fail to reject the null hypothesis. Thus, we don't have sufficient evidence to suggest the alternative hypothesis( proportion of counties with a median education level of high school diploma is diffrent from 45%.).

### 9e.  [1 pt] Now we would like to make a 90% confidence interval for the proportion of ALL U.S. counties that have a median education level of high school diploma.. First, check the conditions for creating this confidence interval.

Condition 1:The observation is independent.
Because the sample is collected via random sampling and n < 10% of population of all US counties.
Condition 2: np >= 10 and n(1-p) >= 10
np = 40 * 45% >= 10, n(1-p) = 40 * (1 - 45%) >= 10.

### 9f.  [2 pt] Now create this 90% confidence interval.

Assume you do not know anything about the population (including $p$).

In [93]:
alpha = 0.10
critical_value = norm.ppf(1-alpha/2)
critical_value

1.6448536269514722

In [96]:
lower_bound=prop - critical_value*(std_error2)
upper_bound=prop + critical_value*(std_error2)
print(lower_bound, "," ,upper_bound)

0.3706147221581398 , 0.6293852778418603


### 9g.  [1 pt] Use this confidence interval to make a conclusion about your hypothesis tests.

Because the null value p = 0.45 is in the confident interval, we fail to reject the null hypothesis. Thus, we don't have sufficient evidence to suggest the alternative hypothesis( proportion of counties with a median education level of high school diploma is diffrent from 45%.).